sim/validator.py

```diff
@@ -0,0 +1,146 @@
+import os
+import subprocess
+import glob
+import json
+import sys
+import argparse
+
+from compare_snps import analyse
+
+DEFAULT_REFERENCE_PATH = './Mycobacterium_bovis_AF212297_LT78304.fa'
+
+def run(cmd, *args, **kwargs):
+    """ Run a command and assert that the process exits with a non-zero exit code.
+        See python's subprocess.run command for args/kwargs
+
+        Parameters:
+            cmd (list): List of strings defining the command, see (subprocess.run in python docs)
+            cwd (str): Set surr
+
+        Returns:
+            None
+    """
+    # TODO: store stdout to a file
+    returncode = subprocess.run(cmd, *args, **kwargs).returncode
+
+    if returncode:
+        raise Exception("""*****
+            %s
+            cmd failed with exit code %i
+          *****""" % (cmd, returncode))
+
+def simulate_genome(reference_path, output_path, num_snps=16000):
+    run([
+        "simuG.pl",
+        "-refseq", reference_path,
+        "-snp_count", str(num_snps),
+        "-prefix", output_path + "simulated"
+    ])
+
+
+def simulate_reads(
+    genome_fasta,
+    output_path,
+    read_length=150,
+    seed=1
+):
+    read_1 = output_path + "simulated_S1_R1_X.fastq"
+    read_2 = output_path + "simulated_S1_R2_X.fastq"
+
+    # TODO: store stdout to a file
+    # TODO: set error statistics
+    run([
+        "wgsim",
+        "-1", str(read_length),
+        "-2", str(read_length),
+        "-S", str(seed),
+        "-r", "0",
+        "-R", "0",
+        "-X", "0",
+        "-e", "0",
+        genome_fasta,
+        read_1,
+        read_2
+    ])
+
+    # TODO: use python library instead?
+    run(["gzip", read_1, read_2])
+
+
+def btb_seq(btb_seq_directory, reads_directory, results_directory):
+    run(["bash", "./btb-seq", reads_directory,
+         results_directory], cwd=btb_seq_directory)
+
+
+def performance_test(btb_seq_path, results_path, reference_path, exist_ok=False):
+    """ Runs a performance test against the pipeline
+
+        Parameters:
+            btb_seq_path (str): Path to btb-seq code is stored
+            results_path (str): Output path to performance test results
+            reference_path (str): Path to reference fasta
+            exist_ok (bool): Whether or not to throw an error if a results directory already exists
+
+        Returns:
+            None
+    """
+    # Validate Input
+    if os.path.isdir(results_path):
+        raise Exception("Output results path already exists")
+    os.makedirs(results_path)
+
+    if not os.path.isdir(btb_seq_path):
+        raise Exception("Pipeline code repository not found")
+
+    # Output Directories
+    simulated_genome_path = results_path + 'simulated-genome/'
```

Comment:
These paths didn't work on my machine -> require leading slash

Although, will need to ensure the user doesn't provide a trailing slash on "results_path"

```suggestion
    simulated_genome_path = results_path + '/simulated-genome/'
```